### PR TITLE
feat(ui): Add ability to archive/restore a release

### PR DESCRIPTION
**Pull request by matejminar**
There are two major problems with how the release deletion works right now:
- it's only possible to delete a release if there are no issues/sessions associated with it (most of the time the trash button is therefore disabled)
- even if the release is deleted, it will get automatically recreated with the next event from that release and therefore cause unexpected behavior

We are introducing the ability to archive release which effectively hides it from the Releases view.

The backend part can be found here: https://github.com/getsentry/sentry/pull/21760

image

image

image

image

### CODE REVIEW

File: src/sentry/static/sentry/app/views/releases/detail/utils.tsx [hunk 1]
```diff
@@ -1,24 +1,72 @@
...
+  addLoadingMessage,
+  addSuccessMessage,
+} from 'app/actionCreators/indicator';
 
 export type CommitsByRepository = {
   [key: string]: Commit[];
 };
 
-export const deleteRelease = (orgId: string, version: string) => {
+export const archiveRelease = async (orgId: string, version: string) => {
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/releases/detail/utils.tsx:25]: Should these be in `app/actionCreators/release` instead? Does the release store need to be updated when releases are archived/unarchived?

File: src/sentry/static/sentry/app/views/releases/detail/releaseActions.tsx [hunk 2]
```diff
@@ -1,65 +1,144 @@
+/* eslint-disable react/prop-types */
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/views/releases/detail/releaseActions.tsx:1]: should we really do that? I noticed that we usually define the `propTypes` when it's necessary.

File: src/sentry/static/sentry/app/views/releases/detail/releaseActions.tsx [hunk 3]
```diff
@@ -1,65 +1,144 @@
...
+        <ProjectsWrapper>{getProjectList()}</ProjectsWrapper>
+
+        {t('Are you sure you want to do this?')}
+      </div>
+    );
+  }
+
+  function getRestoreModalMessage() {
+    return (
+      <div>
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/views/releases/detail/releaseActions.tsx:88]: should this maybe be a React.Fragment?

File: src/sentry/static/sentry/app/views/releases/detail/releaseActions.tsx [hunk 4]
```diff
@@ -1,65 +1,144 @@
...
+
+  function getModalHeader(title: React.ReactNode) {
+    return (
+      <h4>
+        <TextOverflow>{title}</TextOverflow>
+      </h4>
+    );
+  }
+
+  function getArchiveModalMessage() {
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/views/releases/detail/releaseActions.tsx:70]: should we maybe unit `getArchiveModalMessage` and `getRestoreModalMessage` and only pass a parameter with a different message like we did in the `getModalHeader` function? they look very similar

File: src/sentry/static/sentry/app/views/releases/list/releaseListDisplayOptions.tsx [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+import ReleaseListDropdown from './releaseListDropdown';
+
+type Props = {
+  selected: string;
+  onSelect: (key: string) => void;
+};
+
+const ReleaseListDisplayOptions = ({selected, onSelect}: Props) => {
+  const options = [
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/views/releases/list/releaseListDisplayOptions.tsx:13]: I would move this out of `ReleaseListDisplayOptions`

File: src/sentry/static/sentry/app/actionCreators/release.tsx [hunk 6]
```diff
@@ -82,3 +88,59 @@ export function getReleaseDeploys(api: Client, params: ParamsGet) {
       });
     });
 }
+
+export function archiveRelease(api: Client, params: ParamsGet) {
+  const {orgSlug, projectSlug, releaseVersion} = params;
+
+  ReleaseActions.loadRelease(orgSlug, projectSlug, releaseVersion);
+  addLoadingMessage(t('Archiving Release...'));
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/actionCreators/release.tsx:96]:
```suggestion
  addLoadingMessage(t('Archiving Release\u2026'));
```

File: src/sentry/static/sentry/app/actionCreators/release.tsx [hunk 7]
```diff
@@ -82,3 +88,59 @@ export function getReleaseDeploys(api: Client, params: ParamsGet) {
...
+
+  return api
+    .requestPromise(`/organizations/${orgSlug}/releases/`, {
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/actionCreators/release.tsx:99]: nit: I noticed that we use ` try-catch ` more often

File: src/sentry/static/sentry/app/actionCreators/release.tsx [hunk 8]
```diff
@@ -82,3 +88,59 @@ export function getReleaseDeploys(api: Client, params: ParamsGet) {
...
+      );
+      throw error;
+    });
+}
+
+export function restoreRelease(api: Client, params: ParamsGet) {
+  const {orgSlug, projectSlug, releaseVersion} = params;
+
+  ReleaseActions.loadRelease(orgSlug, projectSlug, releaseVersion);
+  addLoadingMessage(t('Restoring Release...'));
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/actionCreators/release.tsx:124]:
```suggestion
  addLoadingMessage(t('Restoring Release\u2026'));
```

**Review by priscilawebdev — APPROVED** [review]
Good Job!!

**Pull request merged**